Sources/TuistCore/Graph/GraphLoader.swift

```diff
@@ -102,19 +103,22 @@ public final class GraphLoader: GraphLoading {
         else {
             throw GraphLoadingError.targetNotFound(name, path)
         }
-
+        
         cache.add(target: target, path: path)
-        let dependencies = try target.dependencies.compactMap { dependency in
-            try loadDependency(
+        let targetDependency = GraphDependency.target(name: name, path: path)
+        let dependencies: [GraphDependency] = try target.dependencies.compactMap { dependency in
+            guard let graphDep = try loadDependency(
                 path: path,
                 forPlatforms: target.supportedPlatforms,
                 dependency: dependency,
                 cache: cache
-            )
+            ) else { return nil }
+            cache.edges[(targetDependency, graphDep)] = dependency.platformFilters
```

Comment:
If I followed along / understood the intent here

- We're using `edges` to store the platform information for each of the graph nodes
  - The alternative would be to associate `platforms` which each of the different node types and propagate this through the code which may be a bit daunting
- We're tracking `platformFilters` for each `source` > `destination` and storing that information under `edges`? (for example `TargetA` depends on `TargetB` and `Target A` supports `iOS`, `edges` will contain an entry `(A,B) = iOS`

There's a few gotchas here:

- This codepath will be skipped if another target depends on `TargetB`  (e.g. `App` > `TargetB`) and we'd end up not having an entry for the platform filters of this dependency
  - This is due to employing a `cache` to optimise the loading process and not load an already loaded node 
- For the filters here are we meaning `supportedPlatforms` instead? as filters would be need to be derived based on the source / destination target's set of platforms (the logic we have adopted in `applyPlatformFilters`?
  - Looks like we're only grabbing `platformFilters` directly from the `destination` here?


Suggestion:

- Would it suffice to only store a map of `node > [Platform]` for a later look up and only the `LinkGenerator` / `BuildPhaseGenerator` would look up the platforms to derive `platformFilters` per the existing logic
  - This may alleviate the traverser code changes 

Overall I'm in two minds about this approach as on one hand I see there's a fair bit to of code changes needed propagate `platforms` to each graph node / dependency, on the other I see this can make crafting a `Graph` in the various tests more tedious and error prone now as we'd need to specify the `Target` `GraphDependency` and remember to include the `platformFilters` separately to match 🤔

Reply:
my bad, I misunderstood the scenario here  

From re-examining - I see we do need edges approach here as there could be different filters for the same dependency 

- `AppA` > `TargetA` (`platformFilters` =  `[iOS]`)
- `AppB` > `TargetA` (`platformFilters` =  `[iOS, macOS]`)

As such the `source` + `destination` pairs are needed to store the filters